app/controllers/master/service_discovery/auth_controller.rb

```diff
@@ -11,7 +11,7 @@ def show
   protected
 
   def self_domain_url(self_domain)
-    options = params.except(:self_domain, :scope).merge(host: self_domain, controller: 'provider/admin/service_discovery/auth')
+    options = params.except(:self_domain, :scope).merge(host: self_domain, controller: 'provider/admin/service_discovery/auth').permit!
```

Comment:
This is a bit risky. Have you tried instead...

```suggestion
    options = params.permit(*%i[code referrer state]).merge(host: self_domain, controller: 'provider/admin/service_discovery/auth')
```
?